scripts/schemas/twister/testsuite-schema.yaml

```diff
@@ -49,6 +49,12 @@ mapping:
       "ignore_qemu_crash":
         type: bool
         required: false
+      "plans":
+        type: seq
+        required: false
+        sequence:
+          - type: str
+            enum: ["smoke", "unit", "integration", "acceptance", "sanity", "functional"]
```

Comment:
> Why such strict list of possible names?

The list can always be expanded. Otherwise we will end up with a mess with custom levels and things will be missed. This is a list of possible levels, does not mean you have to use them all, what fall under each of those is more or less well defined in the testing universe. 

This change has been there for 3 months and the list was open for discussion, but nobody complained. We can modify it expand on it, I am fine with that, but we need to have this pre-defined to avoid random levels going in that will be difficult to manage later.

pr, push, schedule are not testing levels, those are events and everyone can decide what to run and when based on their needs.

The confusion here is coming most likely from plan vs level, I will change everything back to level as it was before, but the schema will have to remain, maybe modifed a bit and following some standard names and what we need as a project.